src/main/cpp/autonomous/AutoDriveForward.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2021 FRC Team 3512. All Rights Reserved.
+
+#include <frc/trajectory/constraint/MaxVelocityConstraint.h>
+#include <frc/trajectory/constraint/RectangularRegionConstraint.h>
+#include <wpi/numbers>
+
+#include "Robot.hpp"
+
+namespace frc3512 {
+
+void Robot::AutoDriveForward() {
+    const frc::Pose2d kInitialPose{12_m, 1.05_m,
+                                   units::radian_t{2 * wpi::numbers::pi}};
+
+    const frc::Pose2d kMidPose{12_m - 1.5 * Drivetrain::kLength, 1.05_m,
+                               units::radian_t{wpi::numbers::pi}};
+
+    const frc::Pose2d kEndPose{10_m, 1.05_m,
+                               units::radian_t{2 * wpi::numbers::pi}};
```

Comment:
If you intended to use the same orientation as the 3 ball drive forward, you should be using pi here instead of 2pi. Also, the trajectory should not be reversed. That breaks things.
```
[ RUN      ] AutonomousTests/AutonomousTest.Run/10
Not loading CameraServerShared
Default RobotInit() method... Override me!
TrajectoryGenerator error: Could not parameterize a malformed spline. This means that you probably had two or more adjacent waypoints that were very close together with headings in opposing directions.
[          ] where GetParam() = "Right Side Drive Forward"
Right Side Drive Forward autonomous
```